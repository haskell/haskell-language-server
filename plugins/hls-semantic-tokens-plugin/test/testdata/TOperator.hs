module TOperator where

-- imported operator
go f x = f $ x
-- operator defined in local module
($$$$) = b
x = 1 $$$$ 2
-- class method take precedence over operator
b = 1 + 1
