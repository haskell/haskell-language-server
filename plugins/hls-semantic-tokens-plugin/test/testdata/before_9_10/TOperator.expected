4:1-3 TFunction "go"
4:4-5 TFunction "f"
4:6-7 TVariable "x"
4:10-11 TFunction "f"
4:11-12 TOperator "$"
4:12-13 TVariable "x"
6:2-6 TOperator "$$$$"
7:1-2 TVariable "x"
7:7-11 TOperator "$$$$"
8:6-7 TTypeVariable "a"
8:8-11 TOperator ":+:"
8:12-13 TTypeVariable "b"
8:16-19 TDataConstructor "Add"
8:20-21 TTypeVariable "a"
8:22-23 TTypeVariable "b"
9:7-10 TOperator ":-:"
9:12-13 TTypeVariable "a"
9:14-15 TTypeVariable "b"
9:19-20 TTypeVariable "a"
9:22-23 TTypeVariable "b"
11:1-4 TFunction "add"
11:8-11 TTypeConstructor "Int"
11:12-15 TOperator ":+:"
11:16-19 TTypeConstructor "Int"
11:23-26 TTypeConstructor "Int"
11:27-30 TOperator ":-:"
11:31-34 TTypeConstructor "Int"
13:1-4 TFunction "add"
13:6-9 TDataConstructor "Add"
13:10-11 TVariable "x"
13:12-13 TVariable "y"
13:18-19 TVariable "x"
13:21-22 TVariable "y"
